### PR TITLE
Trm 21676/add sequence section

### PR DESCRIPTION
**Pull request by xwatkins**
Adding the sequence section to display the canonical and isoform sequences.
- canonical sequence is displayed by default
- isoform sequences are loaded on demand (see react hook usage)

Caveats:
- need to discuss styling with UX (button level definition, potentially horizontal `InfoList` component)
- enzyme doesn't support `hooks` (yes?) so the interaction with button is not tested...
- I'm wondering about having the entry type definition in a separate file reused by all viewers.

### CODE REVIEW

File: src/model/SequenceView.tsx [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+      molWeight: number;
+      crc64: string;
+    };
+    comments?: AlternativeProduct[];
+  };
+};
+
+export const SequenceInfo: React.FC<{
+  isoformId: string;
+  isoformSequence?: SequenceViewProps['data']['sequence'];
```
Comment (dlrice) [inline src/model/SequenceView.tsx]: Could be read easier if `sequence` was it's on type and is used to define `SequenceInfo`'s `isoformSequence` and just added to `SequenceViewProps` ie:
```
SequenceViewProps = {
...
sequence?: sequenceType;
...
```

File: src/model/SequenceView.tsx [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+  const [data, setData] = useState(null);
+  const [isoformToFetch, setIsoformToFetch] = useState('');
+
+  useEffect(() => {
+    if (!isoformToFetch) {
+      return;
+    }
+    const fetchIsoformData = async () => {
+      const result = await fetchData(`${apiUrls.entry(isoformToFetch)}`);
+      setData(result.data.sequence);
```
Comment (dlrice) [inline src/model/SequenceView.tsx]: Would check for existence of the path `result.data.sequence` before accessing it.

File: src/model/SequenceView.tsx [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+}) => {
+  const infoListData = [
+    {
+      title: 'Name',
+      content: isoformData.name.value,
+    },
+    {
+      title: 'Synonyms',
+      content:
+        isoformData.synonyms &&
```
Comment (dlrice) [inline src/model/SequenceView.tsx]: Could use `idx` since it's already imported.

File: src/model/SequenceView.tsx [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+        isoformData.synonyms.map(syn => syn.value).join(', '),
+    },
+    {
+      title: 'Note',
+      content:
+        isoformData.note &&
+        isoformData.note.texts.map(note => note.value).join(', '),
```
Comment (dlrice) [inline src/model/SequenceView.tsx]: Could use `idx` since it's already imported.

File: src/model/SequenceView.tsx [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+  }
+  return (
+    <Fragment>
+      <InfoList infoData={sequenceInfoData} />
+      <p>
+        This entry describes{' '}
+        <strong>{alternativeProducts.isoforms.length}</strong> isoforms produced
+        by <strong>{alternativeProducts.events.join(' & ')}</strong>.
+      </p>
+      <p>{alternativeProducts.note.texts.map(text => text.value).join(' ')}</p>
```
Comment (dlrice) [inline src/model/SequenceView.tsx]: Need some existence checks here in the path `alternativeProducts.note.texts`. An error can be observed at entry `Q9HC29`.

File: src/model/SequenceView.tsx [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+      {alternativeProducts.isoforms.map(isoform => {
+        let isoformSequence, lastUpdateDate;
+        // Canonical
+        if (isoform.isoformSequenceStatus === 'Displayed') {
+          isoformSequence = data.sequence;
+          if (data.entryAudit) {
+            lastUpdateDate = `${data.entryAudit.lastSequenceUpdateDate} v${
```
Comment (dlrice) [inline src/model/SequenceView.tsx]: Is all of `data.entryAudit.lastSequenceUpdateDate` guaranteed to exist?

File: src/model/SequenceView.tsx (hunk 3 above)
Comment (xwatkins) [reply src/model/SequenceView.tsx]: > If an intermediate property is either `null` or `undefined`, it is instead returned. The purpose of this function is to simplify extracting properties from a chain of maybe-typed properties.

We need to check that the map is not `null` or `undefined` anyway so I don't think `idx` gives us anything here?

File: src/model/SequenceView.tsx (hunk 4 above)
Comment (xwatkins) [reply src/model/SequenceView.tsx]: Same as above

File: src/model/SequenceView.tsx (hunk 1 above)
Comment (xwatkins) [reply src/model/SequenceView.tsx]: Done

File: src/model/SequenceView.tsx (hunk 5 above)
Comment (xwatkins) [reply src/model/SequenceView.tsx]: Good spot. Sorted

File: src/model/SequenceView.tsx (hunk 6 above)
Comment (xwatkins) [reply src/model/SequenceView.tsx]: It is if `entryAudit` is present. I've changed the type def to reflect this.

File: src/model/SequenceView.tsx (hunk 1 above)
Comment (dlrice) [reply src/model/SequenceView.tsx]: `isoformSequence?: SequenceViewProps['data']['sequence'];` can use the new type you've made to become `isoformSequence?: Sequence;`

Comment (xwatkins) [reply src/model/SequenceView.tsx]: d'oh! done.

**Pull request merged**